azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/utils/MacUtils.java

```diff
@@ -0,0 +1,139 @@
+                 final BufferedReader br = new BufferedReader(inputStreamReader)) {
+                String tmp;
+                while ((tmp = br.readLine()) != null) {
+                    ret.append(tmp);
+                }
+            }
+            if (process.waitFor() != 0) {
+                throw new IOException(String.format("Command %s execute fail.", String.join(" ", command)));
+            }
+        } catch (IOException | InterruptedException ex) {
+            return macs;
+        }
+        String commandMacsString = ret.toString();
+
+        Matcher matcher = MAC_PATTERN.matcher(commandMacsString);
+        while (matcher.find()) {
+            String mac = matcher.group(0);
+            if (isValidMac(mac)) {
+                macs.add(mac);
+            }
+        }
+        return macs;
+    }
+
+    public static String getMacByNetworkInterface() {
+        List<String> macs = getMacsByNetworkInterface();
+        if (CollectionUtils.isEmpty(macs)) {
+            return StringUtils.EMPTY;
+        }
+        return macs.get(0);
+    }
+
+    private static List<String> getMacsByNetworkInterface() {
+        List<String> macs = new ArrayList<>();
+        try {
+            Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+            while (interfaces.hasMoreElements()) {
+                NetworkInterface networkInterface = interfaces.nextElement();
+                if (networkInterface.isLoopback()) {
+                    continue;
+                }
+                if (networkInterface.getHardwareAddress() != null) {
+                    byte[] mac = networkInterface.getHardwareAddress();
+                    // Refers https://www.mkyong.com/java/how-to-get-mac-address-in-java/
+                    StringBuilder sb = new StringBuilder();
+                    for (int i = 0; i < mac.length; i++) {
+                        sb.append(String.format("%02X%s", mac[i], (i < mac.length - 1) ? "-" : ""));
+                    }
+                    String macStr = sb.toString();
+                    if (isValidMac(macStr)) {
+                        macs.add(macStr);
+                    }
+                }
+            }
+        } catch (SocketException e) {
+            return macs;
+        }
+        return macs;
+    }
+
+    private static boolean isValidMac(String mac) {
+        if (StringUtils.isEmpty(mac)) {
+            return false;
+        }
+        if (!isValidRawMac(mac)) {
+            return false;
+        }
+        final String fixedMac = mac.replaceAll("-", ":");
+        return Arrays.stream(INVALID_MAC_ADDRESS).noneMatch(invalidateMacAddress -> StringUtils.equalsIgnoreCase(fixedMac, invalidateMacAddress));
+    }
+
+    private static boolean isValidRawMac(String raw) {
+        return StringUtils.isNotEmpty(raw) && INTACT_MAC_PATTERN.matcher(raw).find();
+    }
+
+    public static void main(String[] args) {
```

Comment:
Please remove the test codes and add unit test cases if needed

Reply:
removed